lib/pdf_wrangling.py

```diff
@@ -0,0 +1,68 @@
+from reportlab.pdfgen import canvas
+from reportlab.lib.colors import magenta, pink, blue, green
+from typing import List
+from pikepdf import Pdf
+import io
+
+class FormField:
+    """A data holding class, used to easily specify how a PDF form field should be created."""
+    def __init__(self, program_name, type_name, x, y, user_name=''):
+        self.type = type_name
+        self.name = program_name
+        self.x = x
+        self.y = y
+        self.user_name = user_name
+
+def _create_only_fields(io_obj, fields_per_page:List[List[FormField]]):
+    """Creates a PDF that contains only AcroForm fields. This PDF is then merged into an existing PDF to add fields to it.
+    We're adding fields to a PDF this way because reportlab isn't able to read PDFs, but is the best feature library for
+    writing them.
+
+    TODO: add the rest of the fields, including nesting multiple fields as one like a radio button,
+    using https://www.reportlab.com/docs/reportlab-reference.pdf
+    """
+    c = canvas.Canvas(io_obj)
+    c.setFont('Courier', 20)
+    form = c.acroForm
+    for fields in fields_per_page:
+        for field in fields:
+            if field.type == 'Tx': # Text input Field
+                # TODO(brycew): just **kwargs the FormField attributes?
+                form.textfield(name=field.name, x=field.x, y=field.y)
```

Comment:
Just noting here we'd want to make sure to have some defaults like no scrollable text, multi-line inputs.

I'm also noting just two coordinates but we'll want a bounding box, right?

Reply:
Good call on the no scrollable defaults, I've added those.

And yep, I also started taking in fields as bounding boxes, lower left X and Y, and then the width and height.